x-pack/test/functional_basic/apps/ml/data_visualizer/index.ts

```diff
@@ -14,9 +14,14 @@ export default function ({ loadTestFile }: FtrProviderContext) {
     );
 
     // The data visualizer should work the same as with a trial license, except the missing create actions
-    // That's why 'index_data_visualizer_actions_panel' is not loaded here
+    // That's why 'index_data_visualizer_actions_panel_trial' is not loaded here
     loadTestFile(
       require.resolve('../../../../functional/apps/ml/data_visualizer/index_data_visualizer')
     );
+    loadTestFile(
+      require.resolve(
+        '../../../../functional/apps/ml/data_visualizer/index_data_visualizer_actions_panel_basic'
```

Comment:
For the `trial` / `basic` split that I suggested, I had a slightly different structure in mind with less code duplication. But reading through your changes, I like your approach even better because it allows us to more easily extend it in the future and it's much clearer. So let's stay with the split as is, I'd just like to move files around a bit:
Since the `index_data_visualizer_actions_panel_basic` file is only used in the basic license tests, we should move it to `x-pack/test/functional_basic/apps/ml/data_visualizer` and can remove the `_basic` suffix. Similarly, we can remove the `_trial` suffix from the other file and then adjust the comment in this index file to something like
```
// That's why the 'basic' version of 'index_data_visualizer_actions_panel' is loaded here
```

Reply:
Updated here https://github.com/elastic/kibana/pull/89471/commits/6119e07af1d0bbc31a85d6db602dc315b26bb38a